utils/wandb_utils.py

```diff
@@ -0,0 +1,89 @@
+
+import wandb
+import os
+import numpy as np
+
+
+def init_wandb(model, config) -> None:
+    """
+    Initialize project on Weights & Biases
+    Args:
+        model (Torch Model): Model for Training
+        args (TrainOptions,optional): TrainOptions class (refer options/train_options.py). Defaults to None.
+    """
+
+    print("Wandb api key provided...")
+    wandb.login(key=config.wandb_id)
+
+    wandb.init(
+        name=config.experiment,
+        project=config.wandb_project,
+        id=config.wandb_id,
+        dir="./",
+    )
+
+    wandb.watch(model, log="all")
+
+
+def wandb_log(train_loss, val_loss, epoch):
+    """
+    Logs the accuracy and loss to wandb
+    Args:
+        train_loss (float): Training loss
+        val_loss (float): Validation loss
+        train_acc (float): Training Accuracy
+        val_acc (float): Validation Accuracy
+        epoch (int): Epoch Number
+    """
+
+    wandb.log({
+        'Loss/Training': train_loss,
+        'Loss/Validation': val_loss,
+    }, step=epoch)
+
+def wandb_save_summary(valid_accuracy,
```

Comment:
Do read about artifacts on wandb as well, for saving the best model and the last checkpoint model to wandb, and this feature as well if possible